### PR TITLE
Adding JSON Plugin

### PR DESCRIPTION
**Pull request by carceneaux**
* Added JSON plugin files
* Added JSON plugin tests
* Added JSON plugin help documentation
* Updated README with JSON plugin example info

I debugged/tested the JSON plugin using several of the example scripts in the project as well as using several reports from the [AsBuiltReport](https://github.com/AsBuiltReport) framework which leverages PScribo.

The purpose of this plugin is to be able to take already existing scripts using PScribo and provide the output in a method that can be ingested by a 3rd party tool. To that end, JSON seemed to be a pretty clear answer as PowerShell can natively do this.

### CODE REVIEW

**Comment by carceneaux** [comment]
Resolved all AppVeyor errors with the exception of one (shown below) as it related to a file not modified in this PR.

CleanShot 2022-12-06 at 16 38 08

I further investigated and it appears to be a newly added rule which is why it didn't pop up before.

**Comment by carceneaux** [comment]
Confirmed. New rule added 2 months ago in the latest PSScriptAnalyzer [release](https://github.com/PowerShell/PSScriptAnalyzer/releases).

CleanShot 2022-12-06 at 16 48 53

**Comment by carceneaux** [comment]
Created a new [issue](https://github.com/iainbrighton/PScribo/issues/123) for this behavior and applied a workaround to patch the issue for now.

**Comment by carceneaux** [comment]
@iainbrighton Thoughts on this?

**Review by iainbrighton — CHANGES_REQUESTED** [review]
You seem to have a good grasp of how the plugin architecture works and, as long as you are happy with the Json structure, it looks pretty good to me. There are only couple of minor comments that may or may not need addressing. Good job!👍

File: appveyor.yml [hunk 1]
```diff
@@ -14,7 +14,8 @@ environment:
 install:
     - ps: Write-Verbose -Message "PowerShell version $($PSVersionTable.PSVersion)" -Verbose
     - ps: Install-Module -Name Pester -MaximumVersion 4.99 -Scope CurrentUser -Force -AllowClobber -Verbose
-    - ps: Install-Module -Name PSSCriptAnalyzer, PSake, VirtualEngine.Build -Scope CurrentUser -Force -AllowClobber -Verbose
+    - ps: Install-Module -Name PSScriptAnalyzer -MaximumVersion 1.20.0 -Scope CurrentUser -Force -AllowClobber -Verbose
+    - ps: Install-Module -Name PSake, VirtualEngine.Build -Scope CurrentUser -Force -AllowClobber -Verbose
```
[the remark this answers is not in the document]
Comment (carceneaux) [reply appveyor.yml]: Change reverted.

File: Tests/Plugins/Json/Out-JsonTable.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+$pluginRoot  = Split-Path -Path $here -Parent;
+$testRoot  = Split-Path -Path $pluginRoot -Parent;
+$moduleRoot = Split-Path -Path $testRoot -Parent;
+Import-Module "$moduleRoot\PScribo.psm1" -Force;
+
+InModuleScope 'PScribo' {
+
+    Describe 'Plugins\Json\Out-JsonTable' {
+
+        $isNix = (($PSVersionTable['PSEdition'] -eq 'Core') -and (-not $IsWindows))
```
[the remark this answers is not in the document]
Comment (carceneaux) [reply Tests/Plugins/Json/Out-JsonTable.Tests.ps1]: Good call out. I investigated this comparing unit test results from PowerShell (5.1.17763.2931) running on Windows and PowerShell Core (7.3.0) running on MacOS and found no differences. As such, I've removed the unnecessary check.

**Comment by carceneaux** [comment]
@iainbrighton Everything should now be good to go. Only failure in the build can be ignored as it's being resolved in #125.

Thanks for your time reviewing this! 😄

**Comment by carceneaux** [comment]
@iainbrighton Any further requests on this? Are we good to merge?

**Comment by carceneaux** [comment]
@iainbrighton Howdy! 😄 I just updated this PR to include the recent merges into the dev branch.

Any chance this PR can be merged?

**Comment by iainbrighton** [comment]
> Any chance this PR can be merged?

@carceneaux If it's merged in its current form, there will be no support for the new `List` functionality. Do you plan on adding this functionality?

Thanks, Iain

**Review by iainbrighton — APPROVED** [review]

**Comment by carceneaux** [comment]
Thanks, Iain! I'll create an issue for the new list functionality and test/add support for it.

**Pull request merged**